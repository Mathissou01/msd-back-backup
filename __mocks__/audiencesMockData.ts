import { GetAudiencesByContractIdDocument } from "../src/graphql/codegen/generated-types";

export const getNoAudienceMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [],
        },
      },
    },
  },
];

export const getSingleAudienceMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [
            {
              id: 1,
              attributes: {
                isActive: true,
                type: "Particuliers",
                __typename: "Audience",
              },
            },
          ],
        },
      },
    },
  },
];

export const getTwoAudiencesMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [
            {
              id: 1,
              attributes: {
                isActive: true,
                type: "Particuliers",
                __typename: "Audience",
              },
            },
            {
              id: 2,
              attributes: {
                isActive: true,
                type: "Collectifs",
                __typename: "Audience",
              },
            },
          ],
        },
      },
    },
  },
];

export const getAllAudiencesMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [
            {
              id: 1,
              attributes: {
                isActive: true,
                type: "Particuliers",
                __typename: "Audience",
              },
            },
            {
              id: 2,
              attributes: {
                isActive: true,
                type: "Collectifs",
                __typename: "Audience",
              },
            },
            {
              id: 3,
              attributes: {
                isActive: true,
                type: "Professionnels",
                __typename: "Audience",
              },
            },
          ],
        },
      },
    },
  },
];
