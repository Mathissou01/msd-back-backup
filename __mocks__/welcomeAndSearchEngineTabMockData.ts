import { GetWelcomeMessageAndSearchEngineBlockTabDocument } from "../src/graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetWelcomeMessageAndSearchEngineBlockTabDocument,
      variables: {
        contractId: "0",
      },
    },
    result: {
      data: {
        contractCustomizations: {
          data: [
            {
              attributes: {
                homepage: {
                  data: {
                    attributes: {
                      searchEngineBlock: {
                        data: {
                          id: "1",
                          attributes: {
                            titleContent: "Titre modifi√©",
                          },
                        },
                      },
                      welcomeMessageBlock: {
                        data: {
                          id: "1",
                          attributes: {
                            showBlock: true,
                            subtitle: "Subtitle",
                            title: "Title",
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          ],
        },
      },
    },
  },
];
