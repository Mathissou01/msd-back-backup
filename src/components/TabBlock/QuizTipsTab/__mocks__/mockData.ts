import { GetQuizAndTipsBlockDocument } from "../../../../graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetQuizAndTipsBlockDocument,
      variables: {
        contractId: "1",
      },
    },
    result: {
      data: {
        contractCustomizations: {
          data: [
            {
              attributes: {
                homepage: {
                  data: {
                    attributes: {
                      quizAndTipsBlock: {
                        data: {
                          id: "1",
                          attributes: {
                            title: "Titre modifi√©",
                            displayBlock: false,
                            displayQuiz: false,
                            quiz: {
                              data: {
                                id: "2",
                                attributes: {
                                  title: "Nom du quiz",
                                  publishedAt: "2022-11-20T02:34:14.535Z",
                                },
                              },
                            },
                            displayTips: false,
                            tips: {
                              data: [],
                            },
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          ],
        },
        services: {
          data: [
            {
              attributes: {
                serviceInstance: [
                  {
                    __typename: "ComponentMsdEditorial",
                    id: "4",
                    editorialServices: {
                      data: [
                        {
                          attributes: {
                            subServiceInstance: [
                              {
                                __typename: "ComponentEditoQuizzesSubService",
                                isActivated: true,
                                quizzes: {
                                  data: [
                                    {
                                      id: "2",
                                      attributes: {
                                        title: "Nom du quiz",
                                        publishedAt: "2022-11-20T02:34:14.535Z",
                                      },
                                    },
                                    {
                                      id: "1",
                                      attributes: {
                                        title: "truc",
                                        publishedAt: "2022-11-18T10:16:08.936Z",
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                        {
                          attributes: {
                            subServiceInstance: [
                              {
                                __typename: "ComponentEditoTipsSubService",
                                isActivated: true,
                                tips: {
                                  data: [
                                    {
                                      id: "5",
                                      attributes: {
                                        title: "nah",
                                        publishedAt: "2022-11-19T00:14:37.403Z",
                                      },
                                    },
                                    {
                                      id: "2",
                                      attributes: {
                                        title: "Titre de l'astuce 1",
                                        publishedAt: "2022-11-19T00:14:08.232Z",
                                      },
                                    },
                                    {
                                      id: "1",
                                      attributes: {
                                        title: "Titre de l'astuce 2",
                                        publishedAt: "2022-11-18T10:17:11.158Z",
                                      },
                                    },
                                    {
                                      id: "3",
                                      attributes: {
                                        title: "Titre de l'astuce 3",
                                        publishedAt: "2022-11-19T00:14:20.490Z",
                                      },
                                    },
                                    {
                                      id: "4",
                                      attributes: {
                                        title: "Titre de l'astuce 4",
                                        publishedAt: "2022-11-19T00:14:30.273Z",
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
            },
          ],
        },
      },
    },
  },
];

export const errorMockData = [
  {
    request: {
      query: GetQuizAndTipsBlockDocument,
      variables: {
        contractId: "1",
      },
    },
    error: new Error("An error occured"),
  },
];
