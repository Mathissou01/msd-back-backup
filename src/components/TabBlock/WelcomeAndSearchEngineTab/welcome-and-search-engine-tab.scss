.c-WelcomeAndSearchEngineTab {
  position: relative;
  display: flex;
  flex-direction: column;
  gap: 24px;
  width: 100%;
  height: 100%;

  &__Title {
    margin-bottom: 18px;
    font-size: 28px;
  }

  &__Form {
    display: flex;
    flex-direction: column;
    gap: 24px;
    width: 100%;
  }

  &__Group {
    display: flex;
    flex-direction: column;
  }

  &__SubGroup {
    display: flex;
    flex-direction: column;
    gap: 10px;

    &_short {
      width: 60%;
    }
  }

  &__Buttons {
    display: flex;
    gap: 16px;
  }
}
