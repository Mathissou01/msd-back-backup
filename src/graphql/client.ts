import {
  ApolloClient,
  InMemoryCache,
  NormalizedCacheObject,
} from "@apollo/client";

let client: ApolloClient<NormalizedCacheObject>;

if (process.env.NEXT_PUBLIC_MOCK === "true") {
  /* With .env variable NEXT_PUBLIC_MOCK=true, return mock graphql client with fake response data */
  client = (await import("./mockedClient")).default(1000);
} else {
  client = await new ApolloClient({
    uri: `${process.env.NEXT_PUBLIC_API_URL}/graphql`,
    cache: new InMemoryCache({
      typePolicies: {
        UploadFolder: {
          merge: true,
        },
        UploadFile: {
          merge: true,
        },
        Contract: {
          merge: true,
        },
        ContractCustomization: {
          merge: true,
        },
        New: {
          merge: true,
        },
        Tip: {
          merge: true,
        },
        RecyclingGuideService: {
          merge: true,
        },
        WasteForm: {
          merge: true,
        },
        WasteFamily: {
          merge: true,
        },
        Flow: {
          merge: true,
        },
        FlowColor: {
          merge: true,
        },
        DropOffMap: {
          merge: true,
        },
        Request: {
          merge: true,
        },
        RequestAggregate: {
          merge: true,
        },
      },
    }),
  });
}

export default client;
