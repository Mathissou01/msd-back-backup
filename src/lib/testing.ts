import { prettyDOM } from "@testing-library/react";

export function removeUnstableHtmlPropertiesForTesting(
  htmlElement: HTMLElement,
  attributeName: string,
) {
  const domHTML = prettyDOM(htmlElement, Infinity).toString();
  if (!domHTML) return undefined;
  const regex = new RegExp(`${attributeName}(.*)"(.*)"`, "g");
  return domHTML.replaceAll(regex, "");
}
