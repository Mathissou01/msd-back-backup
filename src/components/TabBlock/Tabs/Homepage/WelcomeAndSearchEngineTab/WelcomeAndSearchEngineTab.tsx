import { FormProvider, useForm } from "react-hook-form";
import { FieldValues } from "react-hook-form/dist/types/fields";
import React, { useState, useEffect } from "react";
import {
  useGetWelcomeMessageAndSearchEngineBlocksByContractIdQuery,
  useUpdateSearchEngineBlockByIdMutation,
  useUpdateWelcomeMessageBlockByIdMutation,
} from "../../../../../graphql/codegen/generated-types";
import {
  extractSearchEngineBlock,
  extractWelcomeMessageBlock,
} from "../../../../../lib/graphql-data";
import { useContract } from "../../../../../hooks/useContract";
import { useFocusFirstElement } from "../../../../../hooks/useFocusFirstElement";
import CommonLoader from "../../../../Common/CommonLoader/CommonLoader";
import CommonButton from "../../../../Common/CommonButton/CommonButton";
import FormInput from "../../../../Form/FormInput/FormInput";
import FormCheckbox from "../../../../Form/FormCheckbox/FormCheckbox";
import "./welcome-and-search-engine-tab.scss";

interface IWelcomeMessageBlock {
  welcomeMessageBlockId: string;
  showBlock: boolean;
  title: string;
  subtitle: string;
}

interface ISearchEngineBlock {
  searchEngineBlockId: string;
  titleContent: string;
}

export default function WelcomeAndSearchEngineTab() {
  /* Static Data */
  const formLabels = {
    welcomeTitle: "Message de bienvenue",
    welcomeShowBlock: "Afficher ce bloc",
    welcomeTitleContent: "Titre",
    welcomeDefaultTitleContent: "Besoin d'aide pour trier un déchet ?",
    welcomeSubtitleContent: "Sous titre",
    welcomeDefaultSubtitleContent:
      "Mon service déchets pour faciliter le tri & la gestion de vos déchets au quotidien",
    searchEngineTitle: "Moteur de recherche",
    searchEngineTitleContent:
      "Texte affiché dans le moteur de recherche principal (haut de page)",
    submitButtonLabel: "Enregistrer les modifications",
    cancelButtonLabel: "Annuler les modifications",
    maxCharactersLabel: "caractères maximum",
  };
  const maxCharactersWelcomeTitle = 40;
  const maxCharactersWelcomeSubtitle = 100;
  const maxCharactersSearchEngine = 30;

  /* Methods */
  async function onSubmitValid(submitData: FieldValues) {
    if (
      submitData["welcomeMessageBlockId"] &&
      submitData["searchEngineBlockId"]
    ) {
      const variablesWelcomeMessageBlock = {
        updateWelcomeMessageBlockId: submitData["welcomeMessageBlockId"],
        data: {
          showBlock: submitData["showBlock"],
          title: submitData["title"],
          subtitle: submitData["subtitle"],
        },
      };
      updateWelcomeMessageBlock({
        variables: variablesWelcomeMessageBlock,
      });

      const variablesSearchEngineBlock = {
        updateSearchEngineBlockId: submitData["searchEngineBlockId"],
        data: {
          titleContent: submitData["titleContent"],
        },
      };
      return updateSearchEngineBlock({
        variables: variablesSearchEngineBlock,
      });
    }
  }

  function onCancel() {
    form.reset();
  }

  /* External Data */
  const { contractId } = useContract();
  const { loading, error, data } =
    useGetWelcomeMessageAndSearchEngineBlocksByContractIdQuery({
      variables: { contractId },
      fetchPolicy: "network-only",
    });
  const [
    updateWelcomeMessageBlock,
    {
      loading: welcomeMessageMutationLoading,
      error: welcomeMessageMutationError,
    },
  ] = useUpdateWelcomeMessageBlockByIdMutation({
    refetchQueries: ["getWelcomeMessageAndSearchEngineBlocksByContractId"],
    awaitRefetchQueries: true,
  });
  const [
    updateSearchEngineBlock,
    { loading: searchEngineMutationLoading, error: searchEngineMutationError },
  ] = useUpdateSearchEngineBlockByIdMutation({
    refetchQueries: ["getWelcomeMessageAndSearchEngineBlocksByContractId"],
    awaitRefetchQueries: true,
  });

  /* Local Data */
  const [welcomeMessageData, setWelcomeMessageData] =
    useState<IWelcomeMessageBlock>();
  const [searchEngineData, setSearchEngineData] =
    useState<ISearchEngineBlock>();
  const formValidationMode = "onChange";
  const form = useForm({
    mode: formValidationMode,
  });
  const { handleSubmit, formState } = form;
  const { isDirty, isSubmitting } = formState;
  const mutationLoading =
    welcomeMessageMutationLoading || searchEngineMutationLoading;
  const mutationError =
    welcomeMessageMutationError || searchEngineMutationError;

  useEffect(() => {
    if (data && data.contractCustomizations?.data) {
      const { welcomeMessageBlock } = extractWelcomeMessageBlock(data);
      if (welcomeMessageBlock?.id && welcomeMessageBlock.attributes) {
        const welcomeMessageBlockData = {
          welcomeMessageBlockId: welcomeMessageBlock.id,
          showBlock: welcomeMessageBlock.attributes.showBlock,
          title: welcomeMessageBlock.attributes.title,
          subtitle: welcomeMessageBlock.attributes.subtitle,
        };
        setWelcomeMessageData(welcomeMessageBlockData);
      }

      const { searchEngineBlock } = extractSearchEngineBlock(data);
      if (searchEngineBlock?.id && searchEngineBlock?.attributes) {
        const searchEngineBlockData = {
          searchEngineBlockId: searchEngineBlock.id,
          titleContent: searchEngineBlock.attributes.titleContent,
        };
        setSearchEngineData(searchEngineBlockData);
      }
    }
  }, [
    form,
    data,
    formLabels.welcomeDefaultTitleContent,
    formLabels.welcomeDefaultSubtitleContent,
  ]);

  useEffect(() => {
    form.reset(Object.assign({}, welcomeMessageData, searchEngineData));
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [welcomeMessageData, searchEngineData]);

  return (
    <div
      className="c-WelcomeAndSearchEngineTab"
      id="panel-welcomeAndSearchEngine"
      role="tabpanel"
      aria-labelledby="tab-welcomeAndSearchEngine"
    >
      <CommonLoader
        isLoading={loading || isSubmitting || mutationLoading}
        isShowingContent={isSubmitting || mutationLoading}
        hasDelay={isSubmitting || mutationLoading}
        errors={[error, mutationError]}
      >
        <FormProvider {...form}>
          <form
            className="c-WelcomeAndSearchEngineTab"
            onSubmit={handleSubmit(onSubmitValid)}
            ref={useFocusFirstElement()}
          >
            <div className="c-WelcomeAndSearchEngineTab__Group">
              <h2 className="c-WelcomeAndSearchEngineTab__Title">
                {formLabels.welcomeTitle}
              </h2>
              <div className="c-WelcomeAndSearchEngineTab__SubGroup c-WelcomeAndSearchEngineTab__SubGroup_short">
                <FormCheckbox
                  name="showBlock"
                  label={formLabels.welcomeShowBlock}
                  isDisabled={mutationLoading}
                />
                <div className="WelcomeAndSearchEngineTab__SubGroup_title">
                  <FormInput
                    type="text"
                    name="title"
                    label={formLabels.welcomeTitleContent}
                    isRequired
                    isDisabled={mutationLoading}
                    maxLengthValidation={maxCharactersWelcomeTitle}
                    validationLabel={`${maxCharactersWelcomeTitle} ${formLabels.maxCharactersLabel}`}
                    defaultValue={welcomeMessageData?.title}
                  />
                </div>
                <div className="WelcomeAndSearchEngineTab__SubGroup_subtitle">
                  <FormInput
                    type="text"
                    name="subtitle"
                    label={formLabels.welcomeSubtitleContent}
                    isRequired
                    isDisabled={mutationLoading}
                    maxLengthValidation={maxCharactersWelcomeSubtitle}
                    validationLabel={`${maxCharactersWelcomeSubtitle} ${formLabels.maxCharactersLabel}`}
                    defaultValue={welcomeMessageData?.subtitle}
                  />
                </div>
              </div>
            </div>
            <div className="c-WelcomeAndSearchEngineTab__Group">
              <h2 className="c-WelcomeAndSearchEngineTab__Title">
                {formLabels.searchEngineTitle}
              </h2>
              <div className="c-WelcomeAndSearchEngineTab__SubGroup c-WelcomeAndSearchEngineTab__SubGroup_short">
                <FormInput
                  type="text"
                  name="titleContent"
                  label={formLabels.searchEngineTitleContent}
                  maxLengthValidation={maxCharactersSearchEngine}
                  validationLabel={`${maxCharactersSearchEngine} ${formLabels.maxCharactersLabel}`}
                  isRequired
                  isDisabled={mutationLoading}
                  defaultValue={searchEngineData?.titleContent}
                />
              </div>
            </div>
            <div className="c-WelcomeAndSearchEngineTab__Buttons">
              <CommonButton
                type="submit"
                label={formLabels.submitButtonLabel}
                style="primary"
                picto="check"
                isDisabled={!isDirty}
              />
              <CommonButton
                type="button"
                label={formLabels.cancelButtonLabel}
                picto="cross"
                onClick={onCancel}
                isDisabled={!isDirty}
              />
            </div>
          </form>
        </FormProvider>
      </CommonLoader>
    </div>
  );
}
