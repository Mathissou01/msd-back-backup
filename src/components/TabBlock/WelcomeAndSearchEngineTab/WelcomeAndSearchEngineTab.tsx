import { FormProvider, useForm } from "react-hook-form";
import { FieldValues } from "react-hook-form/dist/types/fields";
import React, { useState, useEffect } from "react";
import {
  GetSearchEngineTabDocument,
  useGetSearchEngineTabQuery,
  useUpdateSearchEngineTabMutation,
} from "../../../graphql/codegen/generated-types";
import { extractSearchEngineBlock } from "../../../lib/graphql-data";
import { useContract } from "../../../hooks/useContract";
import { useFocusFirstElement } from "../../../hooks/useFocusFirstElement";
import CommonLoader from "../../Common/CommonLoader/CommonLoader";
import CommonButton from "../../Common/CommonButton/CommonButton";
import FormInput from "../../Form/FormInput/FormInput";
import "./welcome-and-search-engine-tab.scss";

interface ISearchEngineBlock {
  id: string;
  titleContent: string;
}

export default function WelcomeAndSearchEngineTab() {
  /* Static Data */
  const formLabels = {
    welcomeTitle: "Message de bienvenue",
    welcomeTitleContent: "Titre",
    welcomeSubtitleContent: "Sous titre",
    searchEngineTitle: "Moteur de recherche",
    searchEngineTitleContent:
      "Texte affiché dans le moteur de recherche principal (haut de page)",
    submitButtonLabel: "Enregistrer les modifications",
    cancelButtonLabel: "Annuler les modifications",
    maxCharactersLabel: "caractères maximum",
  };
  const maxCharacters = 30;

  /* Methods */
  async function onSubmitValid(submitData: FieldValues) {
    if (submitData["id"]) {
      const variables = {
        updateSearchEngineBlockId: submitData["id"],
        data: {
          titleContent: submitData["titleContent"],
        },
      };
      return updateSearchEngineBlock({
        variables,
        refetchQueries: [
          {
            query: GetSearchEngineTabDocument,
            variables: { contractId },
          },
          "getSearchEngineTab",
        ],
      });
    }
  }

  function onCancel() {
    form.reset();
  }

  /* External Data */
  const { contractId } = useContract();
  const { loading, error, data } = useGetSearchEngineTabQuery({
    variables: { contractId },
  });
  const [
    updateSearchEngineBlock,
    { loading: mutationLoading, error: mutationError },
  ] = useUpdateSearchEngineTabMutation();

  /* Local Data */
  const [searchEngineData, setSearchEngineData] =
    useState<ISearchEngineBlock>();
  const formValidationMode = "onChange";
  const form = useForm({
    mode: formValidationMode,
  });
  const { handleSubmit, formState } = form;
  const { isDirty, isSubmitting } = formState;

  useEffect(() => {
    if (data && data.contractCustomizations?.data) {
      const { searchEngineBlock } = extractSearchEngineBlock(data);
      if (searchEngineBlock?.id && searchEngineBlock?.attributes) {
        const mappedData = {
          id: searchEngineBlock.id,
          titleContent: searchEngineBlock.attributes.titleContent,
        };
        setSearchEngineData(mappedData);
        form.reset(mappedData);
      }
    }
  }, [form, data]);

  return (
    <div
      className="c-WelcomeAndSearchEngineTab"
      id="panel-welcomeAndSearchEngine"
      role="tabpanel"
      aria-labelledby="tab-welcomeAndSearchEngine"
    >
      <CommonLoader
        isLoading={loading || isSubmitting || mutationLoading}
        isShowingContent={isSubmitting || mutationLoading}
        hasDelay={isSubmitting || mutationLoading}
        errors={[error, mutationError]}
      >
        <FormProvider {...form}>
          <form
            className="c-WelcomeAndSearchEngineTab"
            onSubmit={handleSubmit(onSubmitValid)}
            ref={useFocusFirstElement()}
          >
            {/*<div className="c-WelcomeAndSearchEngineTab__Group">*/}
            {/*  <h2 className="c-WelcomeAndSearchEngineTab__Title">*/}
            {/*    {formLabels.welcomeTitle}*/}
            {/*  </h2>*/}
            {/*  <div className="c-WelcomeAndSearchEngineTab__SubGroup">*/}
            {/*    <span>[...]</span>*/}
            {/*  </div>*/}
            {/*</div>*/}
            <div className="c-WelcomeAndSearchEngineTab__Group">
              <h2 className="c-WelcomeAndSearchEngineTab__Title">
                {formLabels.searchEngineTitle}
              </h2>
              <div className="c-WelcomeAndSearchEngineTab__SubGroup c-WelcomeAndSearchEngineTab__SubGroup_short">
                <FormInput
                  type="text"
                  name="titleContent"
                  label={formLabels.searchEngineTitleContent}
                  maxLengthValidation={maxCharacters}
                  validationLabel={`${maxCharacters} ${formLabels.maxCharactersLabel}`}
                  isRequired={true}
                  isDisabled={mutationLoading}
                  defaultValue={searchEngineData?.titleContent}
                />
              </div>
            </div>
            <div className="c-WelcomeAndSearchEngineTab__Buttons">
              <CommonButton
                type="submit"
                label={formLabels.submitButtonLabel}
                style="primary"
                picto="check"
                isDisabled={!isDirty}
              />
              <CommonButton
                type="button"
                label={formLabels.cancelButtonLabel}
                picto="cross"
                onClick={onCancel}
                isDisabled={!isDirty}
              />
            </div>
          </form>
        </FormProvider>
      </CommonLoader>
    </div>
  );
}
