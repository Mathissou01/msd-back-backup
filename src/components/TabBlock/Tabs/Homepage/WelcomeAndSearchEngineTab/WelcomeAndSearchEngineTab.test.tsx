import { render, screen } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import React from "react";
import { defaultMockData } from "../../../../../../__mocks__/welcomeAndSearchEngineTabMockData";
import WelcomeAndSearchEngineTab from "./WelcomeAndSearchEngineTab";

jest.mock("../../../../../graphql/client", () => null);

const showBlockTestId = "form-checkbox";
const welcomeTitleValue = "Title";
const welcomeSubtitleValue = "Subtitle";
const searchEngineTitleValue = "Titre modifiÃ©";

describe("WelcomeAndSearchEngineTab", () => {
  it("renders loading and loaded state", async () => {
    const { container } = render(
      <MockedProvider mocks={defaultMockData} addTypename={false}>
        <WelcomeAndSearchEngineTab />
      </MockedProvider>,
    );

    expect(await screen.findByTestId("common-spinner")).toBeInTheDocument();
    expect(await container).toMatchSnapshot();
    expect(await screen.findByTestId(showBlockTestId)).toBeInTheDocument();
    expect(
      await screen.findByDisplayValue(welcomeTitleValue),
    ).toBeInTheDocument();
    expect(
      await screen.findByDisplayValue(welcomeSubtitleValue),
    ).toBeInTheDocument();
    expect(
      await screen.findByDisplayValue(searchEngineTitleValue),
    ).toBeInTheDocument();
    expect(await container).toMatchSnapshot();
  });
});
