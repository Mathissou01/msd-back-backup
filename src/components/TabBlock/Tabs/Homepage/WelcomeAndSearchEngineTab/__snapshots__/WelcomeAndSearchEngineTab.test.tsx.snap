// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeAndSearchEngineTab renders loading and loaded state 1`] = `
<div>
  <div
    aria-labelledby="tab-welcomeAndSearchEngine"
    class="c-WelcomeAndSearchEngineTab"
    id="panel-welcomeAndSearchEngine"
    role="tabpanel"
  >
    <div
      class="c-CommonLoader c-CommonLoader_cover c-CommonLoader_grow"
    >
      <div
        class="c-CommonSpinner"
        data-testid="common-spinner"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;

exports[`WelcomeAndSearchEngineTab renders loading and loaded state 2`] = `
<div>
  <div
    aria-labelledby="tab-welcomeAndSearchEngine"
    class="c-WelcomeAndSearchEngineTab"
    id="panel-welcomeAndSearchEngine"
    role="tabpanel"
  >
    <form
      class="c-WelcomeAndSearchEngineTab"
    >
      <div
        class="c-WelcomeAndSearchEngineTab__Group"
      >
        <h2
          class="c-WelcomeAndSearchEngineTab__Title"
        >
          Message de bienvenue
        </h2>
        <div
          class="c-WelcomeAndSearchEngineTab__SubGroup c-WelcomeAndSearchEngineTab__SubGroup_short"
        >
          <div
            class="c-FormCheckbox"
          >
            <input
              class="c-FormCheckbox__Input "
              data-testid="form-checkbox"
              disabled=""
              id="showBlock"
              name="showBlock"
              type="checkbox"
            />
            <div
              class="c-FormLabel"
            >
              <label
                class="c-FormLabel__LabelWrapper"
                for="showBlock"
              >
                <span
                  class="c-FormLabel__Label"
                >
                  Afficher ce bloc
                </span>
              </label>
            </div>
          </div>
          <div
            class="WelcomeAndSearchEngineTab__SubGroup_title"
          >
            <div
              class="c-FormInput"
            >
              <div
                class="c-FormLabel"
              >
                <label
                  class="c-FormLabel__LabelWrapper"
                  for="title"
                >
                  <span
                    class="c-FormLabel__Label"
                  >
                    Titre *
                  </span>
                  <span
                    class="c-FormLabel__Validation"
                  >
                    40 caractères maximum
                  </span>
                </label>
                <div
                  class="c-FormLabel__Content"
                >
                  <input
                    aria-errormessage="title_error"
                    aria-invalid="false"
                    class="c-FormInput__Input"
                    data-testid="form-input"
                    disabled=""
                    id="title"
                    maxlength="40"
                    name="title"
                    type="text"
                    value="Title"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="WelcomeAndSearchEngineTab__SubGroup_subtitle"
          >
            <div
              class="c-FormInput"
            >
              <div
                class="c-FormLabel"
              >
                <label
                  class="c-FormLabel__LabelWrapper"
                  for="subtitle"
                >
                  <span
                    class="c-FormLabel__Label"
                  >
                    Sous titre *
                  </span>
                  <span
                    class="c-FormLabel__Validation"
                  >
                    100 caractères maximum
                  </span>
                </label>
                <div
                  class="c-FormLabel__Content"
                >
                  <input
                    aria-errormessage="subtitle_error"
                    aria-invalid="false"
                    class="c-FormInput__Input"
                    data-testid="form-input"
                    disabled=""
                    id="subtitle"
                    maxlength="100"
                    name="subtitle"
                    type="text"
                    value="Subtitle"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c-WelcomeAndSearchEngineTab__Group"
      >
        <h2
          class="c-WelcomeAndSearchEngineTab__Title"
        >
          Moteur de recherche
        </h2>
        <div
          class="c-WelcomeAndSearchEngineTab__SubGroup c-WelcomeAndSearchEngineTab__SubGroup_short"
        >
          <div
            class="c-FormInput"
          >
            <div
              class="c-FormLabel"
            >
              <label
                class="c-FormLabel__LabelWrapper"
                for="titleContent"
              >
                <span
                  class="c-FormLabel__Label"
                >
                  Texte affiché dans le moteur de recherche principal (haut de page) *
                </span>
                <span
                  class="c-FormLabel__Validation"
                >
                  30 caractères maximum
                </span>
              </label>
              <div
                class="c-FormLabel__Content"
              >
                <input
                  aria-errormessage="titleContent_error"
                  aria-invalid="false"
                  class="c-FormInput__Input"
                  data-testid="form-input"
                  disabled=""
                  id="titleContent"
                  maxlength="30"
                  name="titleContent"
                  type="text"
                  value="Titre modifié"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c-WelcomeAndSearchEngineTab__Buttons"
      >
        <button
          class="c-CommonButton c-CommonButton_disabled c-CommonButton_primary c-CommonButton_fontSmall"
          disabled=""
          type="submit"
        >
          <div
            class="c-CommonButton__Picto c-CommonButton__Picto_check"
          />
          Enregistrer les modifications
        </button>
        <button
          class="c-CommonButton c-CommonButton_disabled c-CommonButton_secondary c-CommonButton_fontSmall"
          disabled=""
          type="button"
        >
          <div
            class="c-CommonButton__Picto c-CommonButton__Picto_cross"
          />
          Annuler les modifications
        </button>
      </div>
    </form>
  </div>
</div>
`;
